experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/editableTree.events.spec.ts

```diff
@@ -0,0 +1,413 @@
+				child: { myInnerString: "initial string in child" },
+			},
+			schema,
+			subtype: "test",
+		});
+
+		const root = factory.create(new MockFluidDataStoreRuntime(), "the tree").root.content;
+
+		let rootBeforeChangeCount = 0;
+		let rootAfterChangeCount = 0;
+		let childBeforeChangeCount = 0;
+		let childAfterChangeCount = 0;
+
+		root.on("beforeChange", (upPath) => {
+			rootBeforeChangeCount++;
+		});
+		root.on("afterChange", (upPath) => {
+			rootAfterChangeCount++;
+		});
+
+		assert.strictEqual(rootBeforeChangeCount, 0);
+		assert.strictEqual(rootAfterChangeCount, 0);
+
+		// Replace existing node - myString; should fire events on the root node.
+		root.boxedMyString.content = "new string";
+
+		assert.strictEqual(rootBeforeChangeCount, 1);
+		assert.strictEqual(rootAfterChangeCount, 1);
+
+		// Add node where there was none before - child; should fire events on the root node.
+		root.boxedChild.content = {
+			myInnerString: "initial string in original child",
+		};
+
+		assert.strictEqual(rootBeforeChangeCount, 2);
+		assert.strictEqual(rootAfterChangeCount, 2);
+
+		root.child.on("beforeChange", (upPath) => {
+			childBeforeChangeCount++;
+		});
+		root.child.on("afterChange", (upPath) => {
+			childAfterChangeCount++;
+		});
+
+		assert.strictEqual(childBeforeChangeCount, 0);
+		assert.strictEqual(childAfterChangeCount, 0);
+
+		// Replace myString in child; should fire events on the child node and the root node.
+		root.child.boxedMyInnerString.content = "new string in original child";
+
+		assert.strictEqual(rootBeforeChangeCount, 3);
+		assert.strictEqual(rootAfterChangeCount, 3);
+		assert.strictEqual(childBeforeChangeCount, 1);
+		assert.strictEqual(childAfterChangeCount, 1);
+
+		// Replace the whole child; should fire events on the root node.
+		root.boxedChild.content = {
+			myInnerString: "initial string in new child",
+		};
+
+		assert.strictEqual(rootBeforeChangeCount, 4);
+		assert.strictEqual(rootAfterChangeCount, 4);
+		// No events should have fired on the old address node.
+		assert.strictEqual(childBeforeChangeCount, 1);
+		assert.strictEqual(childAfterChangeCount, 1);
+
+		// Replace myInnerString in new child node; should fire events on the root node (but not on the old child node)
+		root.child.boxedMyInnerString.content = "new string in new child";
+
+		assert.strictEqual(rootBeforeChangeCount, 5);
+		assert.strictEqual(rootAfterChangeCount, 5);
+		// No events should have fired on the old address node.
+		assert.strictEqual(childBeforeChangeCount, 1);
+		assert.strictEqual(childAfterChangeCount, 1);
+
+		// Delete node - myOptionalNumber; should fire events on the root node
+		root.boxedMyOptionalNumber.content = undefined;
+
+		assert.strictEqual(rootBeforeChangeCount, 6);
+		assert.strictEqual(rootAfterChangeCount, 6);
+	});
+
+	it("fire in the expected order and always together", () => {
+		// TODO: once assignment to properties is implemented in EditableTree2, update this test to apply changes like
```

Comment:
This support was merged in, so you should be able to use the setter properties here now.

Reply:
Awesome, will update them